app/domain/authentication/authn_gce/validate_resource_restrictions.rb

```diff
@@ -0,0 +1,66 @@
+module Authentication
+  module AuthnGce
+
+    # This class is responsible for validating resource restrictions which are configured on a Conjur host or user
+    # against google JWT token.
+    ValidateResourceRestrictions = CommandClass.new(
+      dependencies: {
+        extract_resource_restrictions:                ExtractResourceRestrictions.new,
+        validate_resource_restrictions_configuration: ValidateResourceRestrictionsConfiguration.new,
+        validate_resource_restrictions_match_jwt:     ValidateResourceRestrictionsMatchJWT.new,
+        logger:                                       Rails.logger
+      },
+      inputs:       %i(authenticator_name account username credentials)
+    ) do
+
+      def call
+        @logger.debug(
+          LogMessages::Authentication::ValidatingResourceRestrictions.new(
+            @username
+          )
+        )
+        extract_resource_restrictions
+        validate_resource_restrictions_configuration
+        validate_resource_restrictions_match_jwt
+        @logger.debug(LogMessages::Authentication::ValidatedResourceRestrictions.new)
+      end
+
+      private
+
+      def extract_resource_restrictions
```

Comment:
Why not just call `resource_restrictions` directly? Is there another reason for this proxy method?

Reply:
yes. we like the `call` method to tell the story of the class. so in this case, the `call` method will be:
```
        extract_resource_restrictions
        validate_resource_restrictions_configuration
        validate_resource_restrictions_match_jwt
```

and it is clear what is happening in the class's logic. so the `extract_resource_restrictions` is there to tell the story and the `resource_restrictions` method is there for memoization.